### ISSUE TITLE
Vendoring related issues

### ISSUE THREAD

**Issue by raitucarp** [dgraph-io/dgraphgoclient#10]
When I run this code:
```go
package main

import (
	"fmt"
        "log"
	"github.com/dgraph-io/dgraphgoclient/graph"
        "google.golang.org/grpc"
        "golang.org/x/net/context"
)

func main() {

	conn, err := grpc.Dial("127.0.0.1:8081", grpc.WithInsecure())
	if err != nil {
		log.Fatal("DialTCPConnection")
	}
	defer conn.Close()

	c := graph.NewDgraphClient(conn)

	// Running a mutation.
	q := `mutation { set {<alice> <follows> <bob> . 
<alice> <name> "Alice" . 
<bob> <name> "Bob" . }}`

	resp, err := c.Query(context.Background(), &graph.Request{Query: q})
	if err != nil {
		log.Fatalf("Error in getting response from server, %s", err)
	}

	// Running a query.
	q = `{me(_xid_: alice) { name _xid_ follows { name _xid_ }  }}`
	resp, err = c.Query(context.Background(), &graph.Request{Query: q})
	if err != nil {
		log.Fatalf("Error in getting response from server, %s", err)
	}
	root := resp.N
	fmt.Printf("Subgraph %+v\n", root)
}
```
I got this error
```
$ go run main.go
# command-line-arguments
./main.go:19: cannot use conn (type *"google.golang.org/grpc".ClientConn) as type *"github.com/dgraph-io/dgraphgoclient/vendor/google.golang.org/grpc".ClientConn in argument to graph.NewDgraphClient
```

**Comment by marclitchfield** [comment]
Trying like heck to figure this out because I'm bumping into the same issue. I'm a bit of a go noob over here, but I've been learning a lot about go's dependency system and the glide package manager in my efforts to get this working. I'm actually having problems building the dgraphgoclient repository itself (I get the same error reported by this issue), and I feel that if I could figure out how to build the repo, I'd be on my way to getting the dgraphgoclient working in my project. I'll post if I figure anything out... or maybe the dgraph team knows what's up!

**Comment by pawanrawal** [comment]
Hey @marclitchfield 

Sorry, I didn't have a chance to look at it. Checking this now. I will get back to you soon.

**Comment by marclitchfield** [comment]
Well it turns out my problem building the dgraphgoclient repo was because I did not have it nested within my GOPATH, and then I learned that "you cannot escape the GOPATH!".  When I built the dgraphgoclient under the GOPATH, it worked fine. So I think that my inability to build the dgraphgoclient was a red herring. I still do encounter the error described above when importing the client into a sample project, which is located within my GOPATH. First I run 

```
go get -v -d
```

Then ```go build``` or ```go run myprogram.go``` will produce the error described above.

**Comment by tisov** [comment]
How's going? I have the same. May refuse from vendor?

**Comment by jdelamar** [comment]
Actually I am only seeing this with DgraphClient so far. Have used vendoring without any issues in other project. It looks like Go doesn't recognize that *"google.golang.org/grpc".ClientConn (specified in my app's import statement) is the same as *"github.com/dgraph-io/dgraphgoclient/vendor/google.golang.org/grpc".ClientConn, in the NewGraph vendoring folder, even though they are exactly the same.

I guess without this we can't use the go Dgraph client driver?

**Comment by pawanrawal** [comment]
@jdelamar You are right about the cause of the issue. We are not vendoring in gRPC anymore into Dgraph. You should be able to build the client now and use it in your project too.

@tisov , @marclitchfield , @raitucarp Sorry guys for taking time to resolve this. Things should be fine now. Let me know if there are any issues. 

It would be best to either 
1. Remove the `dgraph` and `dgraphgoclient` repo and do a fresh `go get` or 
2. Do a git pull on master. for both the repos. 

Also update your version of gRPC to the latest by running `go get -u google.golang.org/grpc`

**Comment by jdelamar** [comment]
@pawanrawal: Excellent. That worked as intended. I can now connect.
Thanks for the fix!

**Comment by tisov** [comment]
@pawanrawal Great! It's working now. Thanks a lot
